### PR TITLE
[Java] make java worker log file prefix (default to java-worker) configurable

### LINKED ISSUE #33787: [Java] make java worker log file prefix (default to java-worker) configurable
### Description

For java worker, it's log file always being prefixed with "java-worker". And in python log_monitor.py, it hardcodes "java-worker*.log" to be polled for new log msg periodically.  Some configs, like log_to_driver and RAY_BACKEND_LOG_LEVEL, don't prevent the log monitor from polling and publishing logs to gcs. To save some CPU cycle and network bandwidth, especially if there is large amount of logs produced from JVM, we can have an option. like a JVM system property, to set log file prefix for java worker instead of hard coded to "java-worker".

### Use case

For RayDP, there are large number of INFO logs generated by Spark executor. The logs are useful but no need to be log-monitored by ray. We can rename ray java worker log file to be prefixed with something else other than 'java-worker'. Then, ray log monitor will not poll logs from the log files.

### PR DESCRIPTION
**Pull request by jiafuzha**
## Why are these changes needed?

For java worker, it's log file always being prefixed with "java-worker". And in python log_monitor.py, it hardcodes "java-worker*.log" to be polled for new log msg periodically. Some configs, like log_to_driver and RAY_BACKEND_LOG_LEVEL, don't prevent the log monitor from polling and publishing logs to gcs. To save some CPU cycle and network bandwidth, especially if there is large amount of logs produced from JVM, we can have an option. like a JVM system property, to set log file prefix for java worker instead of hard coded to "java-worker".

## Related issue number

Closes #33787 

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by zjf2012** [comment]
@jovany-wang @kira-lin please help review.

**Comment by simon-mo** [comment]
This is going to break log monitor, please make the corresponding change there as well:

https://github.com/ray-project/ray/blob/af759eb7c99079be63c523b869178eb76b65c4ec/python/ray/_private/log_monitor.py#L200-L202

**Comment by zjf2012** [comment]
> This is going to break log monitor, please make the corresponding change there as well:
> 
> https://github.com/ray-project/ray/blob/af759eb7c99079be63c523b869178eb76b65c4ec/python/ray/_private/log_monitor.py#L200-L202

As tested, log monitor worked well with either property "ray.logging.file-prefix" set to "raydp-java-worker" or unset and default to "java-worker". 

Could you please be more specific why log_monitor is breaking? From code, 'glob.glob(f"{self.logs_dir}/java-worker*.log"' returns empty list if there is not java-worker* files.

thanks

**Comment by jovany-wang** [comment]
@zjf2012 So your purpose is to prevent polling java worker log messages to GCS right?

**Comment by zjf2012** [comment]
> @zjf2012 So your purpose is to prevent polling java worker log messages to GCS right?

exactly. And it's configurable.

File: java/runtime/src/main/resources/ray.default.conf
Comment (jovany-wang) [inline java/runtime/src/main/resources/ray.default.conf:92]: Add comment on what this field is.

**Review by jovany-wang — APPROVED** [review]
LGTM.

**Comment by jovany-wang** [comment]
Hi @simon-mo , I don't think this breaks anything. Maybe also @rkooo567  have a double check.

**Comment by jovany-wang** [comment]
DO NOT MERGE until 2.4 branch cut.

**Comment by jovany-wang** [comment]
@zjf2012 do you need this feature in Ray2.4?

**Comment by simon-mo** [comment]
Ah. This is a hacky way to not to stream log through log monitor it seems. Alternatively an env var in log monitor yo flag this will do.

**Comment by zjf2012** [comment]
> @zjf2012 do you need this feature in Ray2.4?

yes, we need it in ray 2.4. 

By the way, why do we need to wait branch 2.4 before merging it?

**Comment by jovany-wang** [comment]
@simon-mo If we don't have the need that preventing polling only Java log, it makes more sense to add a flag to logmonitor.

**Comment by jovany-wang** [comment]
@zjf2012 Because we changed the releasing process. We shouldn't merge unnecessary changes when in the process of releasing duration.



> yes, we need it in ray 2.4.

@zhe-thoughts CC

**Comment by zhe-thoughts** [comment]
I'd like @simon-mo to confirm about his open question first. Thanks for working on this @jovany-wang

**Comment by simon-mo** [comment]
With the comments added, the change LGTM

**Comment by jovany-wang** [comment]
Thanks for the contribution! This will be in the next release.

**Pull request merged**
